### PR TITLE
DROOLS-5874 : Guided Rule Editor: Call method on XYZ with formula containing quotes does not re-open correctly

### PR DESCRIPTION
**Pull request by Rikkola**
**JIRA**: 

[link](https://issues.redhat.com/browse/DROOLS-5874)
 

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java [hunk 1]
```diff
@@ -1202,6 +1202,39 @@ public void testDotInString() {
...
+        String drl = "import java.lang.Number;\n"
+                + "import java.util.ArrayList;\n"
+                + "rule rule1\n"
+                + "when\n"
+                + "a : ArrayList( )\n"
+                + "bar : Number( )\n"
+                + "foo : Number( )\n"
+                + "then\n"
+                + "a.add( \"\\\"foo, bar\\\"\" );\n"
+                + "end";
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:1216]: @Rikkola @jstastny-cz could you please help me in understanding this complex quoting? 
- Does the method `add` take one argument of type `String`?
- Lets say the `when` part of the rule was evaluated to `true` and lets say variable `bar` holds value `1` and the variable `foo` holds value `2`, I am correct that we would like to produce java code, that will add one item into list  `a`, and this item will be `"2, 1"`?

If the answer for both is true, I think such value composed from two bound variables can be created via formula expression just as: `foo ", " + bar`

Could you please explain me more properly steps to reproduce?

Comment (jstastny-cz) [reply drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:1216]: I think the breaking input here was a formula using just literals, we've discussed also in the cherry-pick PR in https://github.com/kiegroup/drools/pull/3227#issuecomment-725976698 .

Comment (jstastny-cz) [reply drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:1216]: Though I think we agreed to create a separate issue for the Literal case. DROOLS-5696 should be considered fixed now, this should be tracked separately.

Comment (jomarko) [reply drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceUnmarshallingTest.java:1216]: @jstastny-cz thank you. I reported new jira for it https://issues.redhat.com/browse/DROOLS-5874

@Rikkola please update jira description to match the proper jira number.

Thank you both for help.

**Review by jomarko — APPROVED** [review]
I can confirm user can pass literal value in the form of `"a,b,c,...n"` (leading and trailing quotes) as argument for call method action.

**Comment by Rikkola** [comment]
@danielzhe Would you mind checking the PR? Guilherme checked the one before, but he is now away.

**Review by danielzhe — APPROVED** [review]
Thank you!

**Comment by Rikkola** [comment]
@jstastny-cz Did you want any improvements or are you happy with the PR? Just checking.

Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Ignored me yesterday

Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
org.kie.karaf.itest.camel.kiecamel.KieCamelBlueprintCommandIntegrationTest.testYoungPerson
org.kie.karaf.itest.camel.kiecamel.KieCamelBlueprintCommandIntegrationTest.testOldPerson
org.kie.karaf.itest.camel.kiecamel.KieCamelBlueprintDTIntegrationTest.testFavouriteCheese
org.kie.karaf.itest.camel.kiecamel.KieCamelBlueprintDTIntegrationTest.testNonFavouriteCheese
org.kie.karaf.itest.camel.kiecamel.KieCamelBlueprintIntegrationTest.testYoungPerson
org.kie.karaf.itest.camel.kiecamel.KieCamelBlueprintIntegrationTest.testOldPerson
org.kie.karaf.itest.camel.kiecamel.KieCamelRemoteIntegrationTest.testExecuteCommand
org.kie.karaf.itest.camel.kiecamel.KieCamelRemoteIntegrationTest.testGetProcessDefinition
org.kie.karaf.itest.camel.kiecamel.KieCamelRemoteIntegrationTest.testStartProcess
org.kie.karaf.itest.camel.kiecamel.KieCamelRemoteIntegrationTest.testListContainers

Jenkins run fdb

**Comment by Rikkola** [comment]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardModelIntegrationTest.testApplyPmmlScorecardInBatch[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:41339/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardModelIntegrationTest.testApplyPmmlScorecard[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:41339/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardMultiModuleIntegrationTest.buildAndDeployTest[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:41339/kie-server-services/services/rest/server'}]
org.kie.wb.test.rest.functional.ProjectIntegrationTest.testDeployProjectSnapshotTwice

Jenkins run fdb

**Comment by Rikkola** [comment]
[2020-12-16T11:06:35.948Z] stderr: fatal: unable to access 'https://github.com/kiegroup/kie-soup.git/': Failed connect to github.com:443; Connection timed out

Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Adding one-line test results to commit status...
Setting status of 674549be808037a4d4b96fc00f085056f6a523ae to FAILURE with url https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/fdb/job/drools-master.fdb/72/ and message: 'Build finished. No test results found.'
Using context: Linux - Full Downstream Build
groovy.lang.MissingPropertyException: No such property: owner for class: java.util.AbstractMap$SimpleImmutableEntry

Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by mbiarnes** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Local test run in kie-wb-distributions passed.

Jenkins run fdb

**Comment by Rikkola** [comment]
Local test run in kie-wb-distributions passed.

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by Ginxo** [comment]
Jenkins do fdb

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers QA changed to jomarko. So we have both QA and technical reviews. Can I get a merge please.

**Comment by mareknovotny** [comment]
jenkins do cdb

**Pull request merged**